src/services/base.service.js

```diff
@@ -216,7 +218,9 @@ const baseUpdate = async (args, { req }, resourceType) => {
   logger.info(`${resourceType} >>> update`);
   checkContentTypeHeader(req.headers);
   const data = req.body;
-
+  if (data.resourceType !== undefined) {
+    checkSupportedResource(data.resourceType);
+  }
```

Comment:
Including this `undefined` guard for the update case actually allows us to omit the resourceType from HTTP PUT requests. I checked the spec to see if it's allowed, and I think the sentence "The request body SHALL be a FHIR Resource" is enough to infer that the JSON content must match the structure of a FHIR resource.

All this is to say that I think we should be deleting this if block and always checking the supported resource type. That way we can get an error message that says that you can't use "undefined" as a resourceType, which makes sense. Check out what happens when you send this request body in 2 cases:

```
PUT http://localhost:3000/4_0_1/Patient/123

{
  "id": "123"
}
```

and 

```
POST http://localhost:3000/4_0_1/Patient

{
  "id": "123"
}
```

Second one throws an error, first one doesn't